code/modules/telesci/gps.dm

```diff
@@ -82,14 +84,16 @@ var/list/SPS_list = list()
 		data["gpstag"] = gpstag
 		data["autorefresh"] = autorefreshing
 		data["location_text"] = get_location_name()
+		data["transmit"] = transmitting
 		var/list/devices = list()
 		for(var/D in get_list())
 			var/obj/item/device/gps/G = D
 			if(src != G)
 				var/device_data[0]
-				device_data["tag"] = G.gpstag
-				device_data["location_text"] = G.get_location_name()
-				devices += list(device_data)
+				if (G.transmitting)
```

Comment:
Don't bother creating the list if you're not going to use it. This means moving `var/device_data[0]` under the if